mood_bot.py

```diff
@@ -0,0 +1,726 @@
+                days_delta = datetime.timedelta(days=days_until_check)
+                mood_date = now + days_delta
+            mood_year = mood_date.year
+            mood_month = mood_date.month
+            mood_day = mood_date.day
+            next_datetime = datetime.datetime(mood_year, mood_month, mood_day, int(mood_hour_minute[0]),
+                                              int(mood_hour_minute[1]), 0)
+        else:
+            next_datetime = 0
+    return next_datetime
+
+
+@mood_bot.command(name='next_check', help='Provides the date and time of the next mood check.')
+async def next_check(context):
+    """This function allows the user to call the mood_time_retriever function to
+    figure out when the next mood check will be."""
+    author = context.author
+    check_datetime = mood_time_retriever(author)
+    if check_datetime == 0:
+        await context.send("There is no mood check scheduled.")
+    elif auto_checker_power_switch == 'OFF':
+        await context.send("There is no upcoming mood check because the auto mood checker is turned off.")
+    else:
+        await context.send(f"Your next mood check will be at {str(check_datetime)}")
+
+
+""" MOOD CHECK FUNCTIONS """
+
+
+async def auto_checker_error(context, power_switch_input):
+    await context.send(auto_check_input_error_message)
+    raise me.InputMoodError(auto_check_input_error_message, power_switch_input)
+
+
+@mood_bot.command(name='auto', help='Turns auto mood check ON/OFF.',
+                  aliases=['auto_check', 'autocheck', 'autochecker', 'auto_checker'])
+async def mood_auto_checker(context, power):
+    """This function waits until the next mood check is scheduled and then activates the mood check function."""
+    global auto_checker_power_switch
+    author = context.author
+    if power.upper() == 'OFF':
+        target_time = 0
+        auto_checker_power_switch = power.upper()
+    elif power.upper() == 'ON':
+        target_time = mood_time_retriever(author)
+        auto_checker_power_switch = power.upper()
+    else:
+        target_time = 0
+        await auto_checker_error(context, power)
+    if isinstance(target_time, datetime.datetime):
+        while datetime.datetime.now() < target_time:
+            if auto_checker_power_switch == 'OFF':
+                break
+            await asyncio.sleep(1)
+        if auto_checker_power_switch == 'ON':
+            await mood_reset(context, False)
+            await mood_check(context)
+
+
+@mood_bot.command(name='status', help='Reports whether mood auto checker is on or off, '
+                                      'whether a mood check is in progress, '
+                                      'and what stage of the mood check process is open if applicable.')
+async def mood_status(context):
+    await context.send(f"The auto mood checker is {auto_checker_power_switch}.")
+    mood_check_status = 'has not been started.'
+    if date_gate == 'Closed':

```

Comment:
this gate variable should be a integer

Reply:
if we have  `stage_array = ['date', 'type', 'rating', 'explanation']`

the user can have a `gate_stage` which is an integer

then, if your gate_stage is at '2', you are doing a rating

Reply:
I have switched the stage variable to an integer, which increments as you stated in the next comment.